dvc_ssh/__init__.py

```diff
@@ -51,24 +49,11 @@ def _prepare_credentials(self, **config):
         login_info["client_factory"] = config.get(
             "client_factory", InteractiveSSHClient
         )
-        try:
-            user_ssh_config = parse_config(host=config["host"])
-        except FileNotFoundError:
-            user_ssh_config = {}
-
-        login_info["host"] = user_ssh_config.get("Hostname", config["host"])
-
-        login_info["username"] = (
-            config.get("user")
-            or config.get("username")
-            or user_ssh_config.get("User")
-            or getpass.getuser()
-        )
-        login_info["port"] = (
-            config.get("port")
-            or silent(int)(user_ssh_config.get("Port"))
-            or DEFAULT_PORT
-        )
+        login_info["host"] = config.get("host")
+        if username := (config.get("user") or config.get("username")):
+            login_info["username"] = username
+        if port := config.get("port"):
+            login_info["port"] = port
 
         for option in ("password", "passphrase"):
             login_info[option] = config.get(option)
```

Comment:
Note that we don't really need `ask_*` options, as we have support for keyboard-interactive authentication support. See https://github.com/iterative/dvc-ssh/issues/12.

But this is not a big issue either. It would be nice to also support password authentication interactively, but that's a nice to have.